src/compiler/checker.ts

```diff
@@ -14544,6 +14544,16 @@ namespace ts {
             }
         }
 
+        function checkExpressionForMutableLocationWithContextualType(next: Expression, sourcePropType: Type) {
+            next.contextualType = sourcePropType;
+            try {

```

Comment:
What's causing a `throw` here?

Reply:
A cancellation token triggered anywhere in the checker. We use this pattern anywhere we set `node.contextualType` now, as we can leak memory in the language service if we do not.